File: Android/indoor_localization/lib/domain/entities/asset_position_history_log.dart
import 'point.dart';

class AssetPositionHistoryLog {
  int id;
  int assetId;
  int facilityId;
  Point position;
  DateTime dateTime;

  AssetPositionHistoryLog({
    required this.id,
    required this.assetId,
    required this.facilityId,
    required this.position,
    required this.dateTime,
  });
}

File: WebApp/indoor-localization/src/entities/AssetPositionHistoryLog.ts
import { Point } from "./Point";

export class AssetPositionHistoryLog {
    id: number;
    assetId: number;
    facilityId: number;
    position: Point;
    dateTime: Date;
  
    constructor(
      id: number,
      assetId: number,
      facilityId: number,
      position: Point,
      dateTime: Date
    ) {
      this.id = id;
      this.assetId = assetId;
      this.facilityId = facilityId;
      this.position = position;
      this.dateTime = dateTime;
    }
  }
  